modules/mod_base/controllers/controller_api.erl

```diff
@@ -180,11 +180,28 @@ to_json(ReqData, Context) ->
 
 process_post(ReqData, Context) ->
     Module = z_context:get(service_module, Context),
-    case Module:process_post(ReqData, Context) of
+    
+    %% If Content-Type is application/json, try to decode content body.
+    Context1 = case wrq:get_req_header("content-type", ReqData) of
+        "application/json" ->
+            {ReqBody, _RD} = wrq:req_body(ReqData),
+            {struct, Body} = mochijson2:decode(ReqBody),
+            
+            lists:foldl(
+                fun({Key, Value}, Context2) ->
+                    z_context:set_q(z_convert:to_list(Key), Value, Context2)
```

Comment:
Does it make sense to use `set_q` for this? My reason is that it then doesn’t matter whether a client submits form data or a JSON content body, as handling either of those responses will be (almost) the same.